host-configs/environment.cmake

```diff
@@ -13,5 +13,59 @@ set(MPIEXEC_EXECUTABLE "$ENV{MPIEXEC}" CACHE PATH "" FORCE)
 set(ENABLE_GTEST_DEATH_TESTS ON CACHE BOOL "" FORCE)
 set(ENABLE_CALIPER ON CACHE BOOL "")
 
+# If not defined as argument, take from the environment...
+if(NOT DEFINED ENABLE_HYPRE)
+  set(ENABLE_HYPRE "$ENV{ENABLE_HYPRE}" CACHE BOOL "" FORCE)
+endif()
+# ... and then check the value.
+if(ENABLE_HYPRE)
+  set(GEOSX_LA_INTERFACE "Hypre" CACHE STRING "" FORCE)
+else()
+  set(ENABLE_HYPRE OFF CACHE BOOL "" FORCE)
+endif()
+
+# Same pattern
+if(NOT DEFINED ENABLE_TRILINOS)
+  set(ENABLE_TRILINOS "$ENV{ENABLE_TRILINOS}" CACHE BOOL "" FORCE)
+endif()
+if(ENABLE_TRILINOS)
+  set(GEOSX_LA_INTERFACE "Trilinos" CACHE STRING "" FORCE)
+else()
+  set(ENABLE_TRILINOS FALSE CACHE BOOL "" FORCE)
+endif()
+
+if( (ENABLE_HYPRE AND ENABLE_TRILINOS) OR (NOT ENABLE_TRILINOS AND NOT ENABLE_HYPRE))
+  MESSAGE(SEND_ERROR "Exactly one of ENABLE_HYPRE and ENABLE_TRILINOS must be defined.")
+  MESSAGE(SEND_ERROR "ENABLE_HYPRE = ${ENABLE_HYPRE}.")
+  MESSAGE(SEND_ERROR "ENABLE_TRILINOS = ${ENABLE_TRILINOS}.")
+endif()
+
+MESSAGE(STATUS "GEOSX_LA_INTERFACE = ${GEOSX_LA_INTERFACE}")
+
+set(ENABLE_CUDA "$ENV{ENABLE_CUDA}" CACHE BOOL "" FORCE)
+if(ENABLE_CUDA)
+
+  set(CMAKE_CUDA_FLAGS "$ENV{CMAKE_CUDA_FLAGS}" CACHE STRING "" FORCE)
+  if(NOT CMAKE_CUDA_FLAGS)
+    set(CMAKE_CUDA_FLAGS "Unused" CACHE STRING "" FORCE)
+  endif()
+
+  set(CUDA_TOOLKIT_ROOT_DIR "$ENV{CUDA_TOOLKIT_ROOT_DIR}" CACHE PATH "" FORCE)
+  if(NOT CUDA_TOOLKIT_ROOT_DIR)
+    set(CUDA_TOOLKIT_ROOT_DIR "/usr/local/cuda" CACHE PATH "" FORCE)
+  endif()
+
+  set(CUDA_ARCH "$ENV{CUDA_ARCH}" CACHE STRING "" FORCE)
+  if(NOT CUDA_ARCH)
+    set(CUDA_ARCH "sm_70" CACHE STRING "" FORCE)
+  endif()
+
+  set(ENABLE_HYPRE_CUDA "$ENV{ENABLE_HYPRE_CUDA}" CACHE BOOL "" FORCE)
+  if(NOT ENABLE_HYPRE_CUDA)
+    set(ENABLE_HYPRE_CUDA OFF CACHE BOOL "" FORCE)
+  endif()
```

Comment:
is this if block redundant

Reply:
Do you mean redundant w.r.t. the `if(ENABLE_CUDA)` it lives in?
I had seen that there was some special care about having `hypre` work with `CUDA`.
And this is what I wanted to reproduce here.

Reply:
Here also, `ENABLE_HYPRE_CUDA` should be `ON` if we use `CUDA` and `HYPRE`, no need to check for the user input?